vars/pipelineMainnetCheckpointTests.groovy

```diff
@@ -0,0 +1,185 @@
+                    String networkName = params.NAME ?: whoStarted
+                    currentBuild.displayName = "${networkName} #${currentBuild.id}"
+                }
+            },
+            'st': { Map vars ->
+                DockerisedVega dockerisedVega = vars.dockerisedVega
+                withEnv([
+                    "SYSTEM_TESTS_DOCKER_IMAGE_TAG=${dockerisedVega.prefix}",
+                ]) {
+                    stage('General setup') {
+                        sh label: 'Create directories', script: """#!/bin/bash -e
+                            mkdir -p "\$(dirname ${pipelineDefaults.art.systemTestsJunit})"
+                            mkdir -p "${pipelineDefaults.art.systemTestsState}"
+                            mkdir -p "${pipelineDefaults.art.systemTestsLogs}"
+                        """
+                    }
+                    stage('build system-tests docker image') {
+                        dir('system-tests/scripts') {
+                            withDockerRegistry(vars.dockerCredentials) {
+                                sh label: 'build system-tests container', script: '''#!/bin/bash -e
+                                    make prepare-test-docker-image
+                                '''
+                            }
+                        }
+                    }
+                    stage('make proto for system-tests') {
+                        dir('system-tests/scripts') {
+                            sh label: 'make proto', script: '''#!/bin/bash -e
+                                make build-test-proto
+                            '''
+                        }
+                    }
+                }
+            }
+        ],
+        mainStage: { Map vars ->
+            DockerisedVega dockerisedVega = vars.dockerisedVega
+            String ip = vars.jenkinsAgentPublicIP
+            Map<String,String> usefulLinks = dockerisedVega.getUsefulLinks(ip)
+            Map<String,Map<String,String>> endpointInfo = dockerisedVega.getEndpointInformation(ip)
+            List parameters = []
+
+            usefulLinks.eachWithIndex { linkName, linkURL, index ->
+                if (index == 0) {
+                    parameters << booleanParam(description: 'Useful links', name: "[${linkName}]: ${linkURL}")
+                } else {
+                    parameters << booleanParam(name: "[${linkName}]: ${linkURL}")
+                }
+            }
+
+            endpointInfo.each { machine, endpoints ->
+                endpoints.eachWithIndex { endpointType, endpoint, index ->
+                    if (index == 0) {
+                        parameters << booleanParam(description: machine, name: "[${endpointType}]: ${endpoint}")
+                    } else {
+                        parameters << booleanParam(name: "[${endpointType}]: ${endpoint}")
+                    }
+                }
+            }
```

Comment:
This part can be removed. It fills in the `parameters` list, which is not used in your pipeline.